server/migrations/20200708183925-subscription-model-upgrade.js

```diff
@@ -0,0 +1,120 @@
+/* eslint-disable quotes */
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, DataTypes) => {
+  // add columns
+    await queryInterface.addColumn('Subscriptions', 'chain_id', { type: DataTypes.STRING, allowNull: true });
+    await queryInterface.addColumn('Subscriptions', 'community_id', { type: DataTypes.STRING, allowNull: true });
+    await queryInterface.addColumn('Subscriptions', 'offchain_thread_id', { type: DataTypes.INTEGER, allowNull: true });
+    await queryInterface.addColumn('Subscriptions', 'offchain_comment_id', { type: DataTypes.INTEGER, allowNull: true });
+    await queryInterface.addColumn('Subscriptions', 'chain_event_type_id', { type: DataTypes.STRING, allowNull: true });
+    await queryInterface.addColumn('Subscriptions', 'chain_entity_id', { type: DataTypes.STRING, allowNull: true });
+
+
+    // for each subscription, create relevant associations
+    const chains = await queryInterface.sequelize.query(`SELECT * FROM "Chains"`);
+    const chainIds = chains[0].map((c) => c.id);
+    const subscriptions = await queryInterface.sequelize.query(`SELECT * FROM "Subscriptions";`);
+    await Promise.all(subscriptions[0].map(async (s) => {
+      const { object_id, category_id, id } = s;
+      const object_id_split = object_id.split(/-|_/); // hyphen or underscore
+      const p_object_id = object_id_split[1];
+      const entity = object_id_split[0];
+      let query;
+      switch (category_id)  {
+        case 'new-thread-creation': // object_id: "kusama"/"edgeware"
+          if (chainIds.includes(object_id)) {
+            query = `UPDATE "Subscriptions" SET chain_id='${object_id}' WHERE id=${id}`;
+          } else {
+            query = `UPDATE "Subscriptions" SET community_id='${object_id}' WHERE id=${id}`;
+          }
+          await queryInterface.sequelize.query(query);
+          break;
+        case 'new-comment-creation':  // object_id: "discussion_123"
+          // councilmotion_0x3dd9c8b50b896089ddf9db75d231bd23d6596822ba6d5480697ed0833b64f077
+          // referendum_0
+          if (entity === 'discussion') {
+            // associate OffchainThread
+            query = `UPDATE "Subscriptions" SET offchain_thread_id=${Number(p_object_id)} WHERE id=${id};`;
```

Comment:
Conversion to `Number` probably not necessary here/elsewhere in queries, as it'll be converted back to the same string anyway.

Reply:
right, that makes sense.

Reply:
There was something weird going on when I removed the `Number`, where it would have an issue with `.trim()` (which I think has to do with the fact that I'm splitting the `object_id`. If it's not a problem that it exists in the codebase, I'm just going to leave the `Number` in as it works as expected.